src/test/java/ch/wisv/areafiftylan/TicketTransferRestIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.RestAssured.when;
+
+/**
+ * Created by beer on 5-1-16.
+ */
+public class TicketTransferRestIntegrationTest extends IntegrationTest{
+    private final String TRANSFER_ENDPOINT = "/tickets/transfer";
+
+    private User ticketReciever;
+    private Ticket ticket;
+
+    @Autowired
+    private TicketRepository ticketRepository;
+
+    @Autowired
+    private OrderService orderService;
+
+    @Before
+    public void initTransferTest(){
+        ticketReciever = makeTicketReceiver();
+        ticket = makeMockTicket();
+    }
+
+    @After
+    public void cleanTransferTest(){
+        ticketRepository.deleteAll();
+        userRepository.delete(ticketReciever);
+    }
+
+    private User makeTicketReceiver(){
+        User receiver = new User("receiver", new BCryptPasswordEncoder().encode("password"), "receiver@gmail.com");
+        receiver.getProfile()
+                .setAllFields("receiver", " of tickets", "GotYaTicket", Gender.MALE, "Money Owner 4", "2826GJ", "Tomorrowland",
+                        "0906-1111", null);
+
+        userRepository.saveAndFlush(receiver);
+
+        return receiver;
+    }
+
+    private Ticket makeMockTicket(){
+        Ticket t = new Ticket(user, TicketType.EARLY_FULL, false, false);
+        t = ticketRepository.saveAndFlush(t);
+
+        return t;
+    }
+
+    @Test
+    public void testAddTransfer_Anon(){
+        Map<String, String> transferRequest = new HashMap<>();
+        transferRequest.put("ticketKey", ticket.getKey());
+        transferRequest.put("goalUsername", ticketReciever.getUsername());
+
+        given()
+        .when()
+                .content(transferRequest).contentType(ContentType.JSON)
+                .post(TRANSFER_ENDPOINT)
+        .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN);
+
+        ticket = ticketRepository.findByKey(ticket.getKey()).orElse(null);
+        if(ticket == null) Assert.fail("Could not refresh ticket");
+
+        Assert.assertTrue(ticket.isLockedForTransfer());
+    }
+
+    @Test
+    public void testAddTransfer_Owner(){
+        addTicketTransfer(user.getUsername(), "password").then().statusCode(HttpStatus.SC_OK);
+
+        ticket = ticketRepository.findByKey(ticket.getKey()).orElse(null);
+        if(ticket == null) Assert.fail("Could not refresh ticket");
+
+        Assert.assertTrue(ticket.isLockedForTransfer());
+    }
+
+    @Test
+    public void testAddTransfer_Receiver(){
+        addTicketTransfer(ticketReciever.getUsername(), "password").then().statusCode(HttpStatus.SC_FORBIDDEN);
+
+        ticket = ticketRepository.findByKey(ticket.getKey()).orElse(null);
+        if(ticket == null) Assert.fail("Could not refresh ticket");
+
+        Assert.assertTrue(ticket.isLockedForTransfer());
```

Comment:
Same here. The ticket was never prepared for transfer, then why should it be locked?